nncf/quantization/algorithms/accuracy_control/ranker.py

```diff
@@ -290,10 +291,14 @@ def _collect_values_for_each_item(self, model: TModel, data_items: Iterable[Any]
         :param data_items: Data items.
         :return: A list that contains a metric for each item from the dataset.
         """
-        metrics = [
-            self._validation_fn(self._algo_backend.prepare_for_inference(model), [data_item]) \
-                for data_item in data_items
-        ]
+        metrics = []
+        # pylint: disable=protected-access
+        indexed_items = zip(data_items._indices, data_items) if data_items._indices else enumerate(data_items)
+        for data_item_idx, data_item in indexed_items:
+            data_provider = DataProvider([data_item], None, indices=[data_item_idx])
+            metric_value = self._validation_fn(self._algo_backend.prepare_for_inference(model), data_provider)
```

Comment:
Why you call `prepare_for_inference` inside loop?

Reply:
Fixed